api/src/modules/renown/kyc-service.ts

```diff
@@ -0,0 +1,201 @@
+import { RuntimeCompositeDefinition } from "@composedb/types";
+import { ComposeClient } from "@composedb/client";
+import { EventSource } from "cross-eventsource";
+import { JsonAsString, AggregationDocument } from '@ceramicnetwork/codecs';
+import { Codec, decode } from "codeco";
+import { fetchEntries } from "./queries";
+import logger from "../../logger";
+import { CeramicPowerhouseVerifiableCredential } from "./types";
+
+export interface KYCServiceOptions {
+  // The URL of the Ceramic node
+  ceramicUrl?: string;
+  definition: RuntimeCompositeDefinition
+}
+
+export class KYCService {
+
+  protected client: ComposeClient;
+
+  protected eventSource: EventSource | null = null;
+
+  protected entries: any[] = [];
+
+  protected codec: Codec<any, any, any>;
+
+  constructor(options: KYCServiceOptions) {
+    const ceramic = options.ceramicUrl || "http://localhost:7007";
+    this.client = new ComposeClient({
+      ceramic,
+      definition: options.definition as RuntimeCompositeDefinition,
+    });
+
+    this.eventSource = new EventSource(`${ceramic}/api/v0/feed/aggregation/documents`)
+    this.codec = JsonAsString.pipe(AggregationDocument)
+  }
+
+  #fetchEntriesFromCeramic = async (first: number, skip: number) => {
```

Comment:
Here it would be fetchAllCredentials. So the code calling the kyc-service doesn't need to know that it uses Ceramic.